cumulus/client/cirrus-executor/src/lib.rs

```diff
@@ -253,29 +267,60 @@ where
 	async fn produce_bundle(self, slot_info: ExecutorSlotInfo) -> Option<BundleResult> {
 		println!("TODO: solve some puzzle based on `slot_info` to be allowed to produce a bundle");
 
-		let transactions = {
-			// selection policy: minimize the transaction equivocation.
-			println!("TODO: once elected, select unseen transactions from the transaction pool");
-			b"some transactions".to_vec()
+		let now = time::Instant::now();
+		// TODO: proper deadline
+		let deadline = now + time::Duration::from_micros(500);
+
+		// TODO: ready at the best number of primary block?
+		let parent_number = self.client.info().best_number;
+		let mut t1 = self.transaction_pool.ready_at(parent_number).fuse();
+		// TODO: proper timeout
+		let mut t2 = futures_timer::Delay::new(time::Duration::from_micros(100)).fuse();
+
+		let mut pending_iterator = select! {
+			res = t1 => res,
+			_ = t2 => {
+				tracing::warn!(
+					block_number,
+					"Timeout fired waiting for transaction pool, proceeding with production.",
+				);
+				self.transaction_pool.ready()
```

Comment:
Can you explain why we even call `.ready_at()` then, what is the difference between those 2 for this use case?

Reply:
This pattern was simply copied from https://github.com/paritytech/substrate/blob/a404abd3b3/client/basic-authorship/src/basic_authorship.rs#L394-L408 :P

Reply:
Sure, still I'd like to know why that particular construction is actually used :slightly_smiling_face: If no one on our team fully understands it, we're in a rough shape.

Reply:
Have a quick look at `ready_at` and `ready`, basically `ready_at` is a subset of `ready` where `ready_at` is guaranteed that the transaction pool was updated since `block_number` while `ready` doesn't have this guarantee.

Reply:
Should be possible to reduce to `.ready()` then as we will check that transactions collected are not in previous bundle and we have bundles completely decoupled from block production.

Reply:
But you still have to check some states(if the transaction is able to pay the fee), right?

Reply:
Yes, but either way we may have more than one bundle produced depending on how many things there is in transaction pool (I think we'll need to have an upper limit on any particular bundle).

Reply:
I don't have a pretty clear vision here, we could revisit it in the future, we're iterating fastly anyway.